### PR TITLE
Updated to 24.09.0

### PR DESCRIPTION
**Pull request by bwingconda**
poppler 24.09.0

**Destination channel:** defaults

### Links

- [PKG-5339](https://anaconda.atlassian.net/browse/PKG-5339)
- [Upstream repository](https://gitlab.freedesktop.org/poppler/poppler)
- [Upstream changelog/diff](https://gitlab.freedesktop.org/poppler/poppler/-/blob/poppler-24.09.0/NEWS?ref_type=tags)

### Explanation of changes:

- Updated version
- Added tests
- Updated Windows patches with current working versions from conda-forge

[PKG-5339]: https://anaconda.atlassian.net/browse/PKG-5339?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -31,9 +33,13 @@ build:
 
 requirements:
   build:
+    - git  # [not win]
+    # setuptools >= 65.0.0 is lacking the msvccompiler module in distutils.
+    # https://github.com/pypa/setuptools/pull/3505
+    - setuptools 63.4.1
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]:
```suggestion
    - setuptools <65.0.0
```
Less restrictive for maintenance. The same applies to all the usages below.

File: recipe/build.sh [hunk 2]
```diff
@@ -30,13 +33,18 @@ fi
 mkdir -p build
 cd build
 
+git clone git://git.freedesktop.org/git/poppler/test
+
```
Comment (skupr-anaconda) [inline recipe/build.sh]: Could you please move it to the `source/url` section of the `meta.yaml`? It will be a more explicit way of downloading stuff

Comment (bwingconda) [reply recipe/build.sh]: Would this still require git to be listed in the requirements?

Comment (skupr-anaconda) [reply recipe/build.sh]: Yes, `git_url` requires git on `unix` not `win`.

**Comment by skupr-anaconda** [comment]
```
  build:
    - git  # [not win]
```
I think it will fix the issue with `git`

**Comment by bwingconda** [comment]
> ```
>   build:
>     - git  # [not win]
> ```
> 
> I think it will fix the issue with `git`

The recipe does currently supply git in the build requirements except on Windows. For my local build, I had to disable git's precomposed unicode support. Is there an easy way to disable tag checks in conda-build. This test repo has no tags so `git describe` is expected to fail.

> fatal: No names found, cannot describe anything.
> fatal: iconv_open(UTF-8,UTF-8-MAC) failed, but needed:
>    precomposed unicode is not supported.
>    If you want to use decomposed unicode, run
>    "git config core.precomposeunicode false"

**Comment by skupr-anaconda** [comment]
`git_rev`? see https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#source-from-git
Or this https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#source-from-multiple-sources

**Comment by bwingconda** [comment]
Yes, I did look through the docs as well as some of the `conda-build` source to see what hidden magic keys might exist for git sources. There does not appear to be anything helpful in bypassing these checks.

**Comment by skupr-anaconda** [comment]
`gcc_toolchain-feedstock/recipe/decompress.sh` makes `git clone` too. So, possibly, it's the only way if we really need those tests

**Comment by bwingconda** [comment]
I think the real problem is the `.gitconfig` we have in use on our OSX runner images. Outside of correcting that, I was able to bypass a git clone by downloading zipped sources from their repo. Should be fine as long as they don't change VCS hosts again.

**Review by skupr-anaconda — APPROVED** [review]

**Review by SC426 — APPROVED** [review]

**Pull request merged**